src/SAWScript/Crucible/LLVM/Override.hs

```diff
@@ -813,19 +815,25 @@ enforceDisjointAllocGlobal ::
   (LLVMAllocGlobal arch, LLVMPtr (Crucible.ArchWidth arch)) ->
   OverrideMatcher (LLVM arch) md ()
 enforceDisjointAllocGlobal sym loc
-  (LLVMAllocSpec _pmut _pty _palign psz _ploc _pfresh, p)
-  (LLVMAllocGlobal _qloc (L.Symbol qname), q) =
+  (LLVMAllocSpec _pmut _pty _palign psz ploc _pfresh, p)
+  (LLVMAllocGlobal qloc (L.Symbol qname), q) =
   do let Crucible.LLVMPointer pblk _ = p
      let Crucible.LLVMPointer qblk _ = q
      c <- liftIO $ W4.notPred sym =<< W4.natEq sym pblk qblk
      let msg =
            "Memory regions not disjoint:"
            ++ "\n  (base=" ++ show (Crucible.ppPtr p) ++ ", size=" ++ showTerm psz ++ ")"
+           ++ "\n  from " ++ ppProgramLoc ploc
            ++ "\n  and "
            ++ "\n  global " ++ show qname ++ " (base=" ++ show (Crucible.ppPtr q) ++ ")"
+           ++ "\n  from " ++ ppProgramLoc qloc
      addAssert c $ Crucible.SimError loc $
        Crucible.AssertFailureSimError msg ""
 
+ppProgramLoc :: W4.ProgramLoc -> String
```

Comment:
this should be in `W4`. Or maybe `ProgramLoc` should derive `Show`

Reply:
`ProgramLoc` does derive `Show`, but I didn't want an error message containing haskell record syntax like `ProgramLoc { plFunction = ..., plSourceLoc = ... }`. A pretty-printing function for `ProgramLoc` would be a nice addition to what4, though; I will open a ticket asking for one.